### ISSUE TITLE
make failed

### ISSUE THREAD

**Issue by magnekofoed** [sitemule/ILEusion#14]
Hi, I tried to make ILEusion - but got an error when the make did CRTPGM PGM(ILEUSION/ileusion)
The CRTPGM PGM(ILEUSION/ileusion) works when i do it manually in a 5250 session.

Here is the log:

login as: magnek
magnek@spbankt.test.sparebank1.no's password:
$ pwd
/home/MAGNEK
$ cd  ..
$ cd alien
$ ls
ILEastic-master  ILEusion-master  noxDB-master
$ cd ILEusion-master
$ ls
LICENSE       headers       readme.md
barryci.json  makefile      src
$ make
system -qi "DLTOBJ OBJ(ILEUSION/*ALL) OBJTYPE(*FILE)"
system -qi "DLTOBJ OBJ(ILEUSION/*ALL) OBJTYPE(*MODULE)"

/bin/sh: 146647 Abort
make: [makefile:65: clean] IOT/Abort trap (ignored)dspli
system -q "CRTLIB ILEUSION TYPE(*PROD) TEXT('ILEusion')"
system -q "CRTRPGMOD MODULE(ILEUSION/ileusion) SRCSTMF('./src/ileusion.rpgle') DBGVIEW(*ALL) REPLACE(*YES)" | grep '*RNF' | grep -v '*RNF7031' | sed  "s!*!ileusion.rpgle: &!"

system -q "CRTRPGMOD MODULE(ILEUSION/actions) SRCSTMF('./src/actions.rpgle') DBGVIEW(*ALL) REPLACE(*YES)" | grep '*RNF' | grep -v '*RNF7031' | sed  "s!*!actions.rpgle: &!"

actions.rpgle: *RNF5409 00   1722 000093  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1783 000154  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1866 000237  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1925 000296  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1927 000298  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1932 000303  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1941 000312  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   1943 000314  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2004 000375  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2060 000431  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2061 000432  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2062 000433  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2092 000463  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2116 000487  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2117 000488  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2136 000507  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00   2137 000508  The prototyped call returns a value which is lost when
actions.rpgle: *RNF5409 00     17 The prototyped call returns a value which is lost when CALLP
system -q "CRTRPGMOD MODULE(ILEUSION/data) SRCSTMF('./src/data.rpgle') DBGVIEW(*ALL) REPLACE(*YES)" | grep '*RNF' | grep -v '*RNF7031' | sed  "s!*!data.rpgle: &!"

data.rpgle: *RNF5409 00   1324 000144  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1329 000149  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1334 000154  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1361 000181  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1372 000192  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1373 000193  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1516 000336  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00   1521 000341  The prototyped call returns a value which is lost when
data.rpgle: *RNF5409 00      8 The prototyped call returns a value which is lost when CALLP
system -q "CRTRPGMOD MODULE(ILEUSION/callfunc) SRCSTMF('./src/callfunc.rpgle') DBGVIEW(*ALL) REPLACE(*YES)" | grep '*RNF' | grep -v '*RNF7031' | sed  "s!*!callfunc.rpgle: &!"

system "CRTCMOD MODULE(ILEUSION/types) SRCSTMF('./src/types.c') DBGVIEW(*ALL) REPLACE(*YES)"
CZS0607: Module TYPES was created in library ILEUSION on 08.12.18 at 17.58.26.
# Basically do nothing..
echo "Adding binding entry jsonxml"
Adding binding entry jsonxml
# Basically do nothing..
echo "Adding binding entry ileastic"
Adding binding entry ileastic
system -qi "CRTBNDDIR BNDDIR(ILEUSION/ileusion)"
system -qi "ADDBNDDIRE BNDDIR(ILEUSION/ileusion) OBJ((*LIBL/jsonxml *SRVPGM *IMMED) (*LIBL/ileastic *SRVPGM *IMMED))"
qsh <<EOF
liblist -a NOXDB
liblist -a ILEASTIC
liblist -a ILEUSION
system -i "CRTPGM PGM(ILEUSION/ileusion) MODULE(ILEUSION/ileusion ILEUSION/ACTIONS ILEUSION/DATA ILEUSION/CALLFUNC ILEUSION/TYPES) BNDDIR(ILEUSION/ILEUSION)"
EOF
make: *** [makefile:23: ileusion.pgm] Error 9
$

**Comment by WorksOfLiam** [comment]
Hi @magnekofoed 

Thanks for giving the build a try. This is something I can take a look into Monday morning.

**Comment by WorksOfLiam** [comment]
Hi @magnekofoed 

Are you able to run `qsh` from an SSH session?

Looks like that might be the issue - though that's a guess because I can build from 3 different servers.

**Comment by WorksOfLiam** [comment]
@magnekofoed 

Were you able to find out whether you could run `qsh` from your shell?

Cheers,
Liam

**Comment by magnekofoed** [comment]
not sure what you mean with "run qsh from your shell" - but yes - i can run qsh from putty -  and it makes no difference on the error i get.

**Comment by WorksOfLiam** [comment]
@magnekofoed 

Can you also confirm that you are using GNU Make (`gmake`) instead of the system shipped `make`?

**Comment by magnekofoed** [comment]
Hi, I've used both gmake and make with Putty - same error. But strangely enough- it works now in 5250 QSH with both gmake and make.
